crates/common/consensus/lean/src/state.rs

```diff
@@ -48,126 +59,362 @@ impl LeanState {
             justified_slots: VariableList::empty(),
 
             justifications_roots: VariableList::empty(),
-            justifications_roots_validators: BitList::with_capacity(0)
-                .expect("Failed to initialize an empty BitList"),
+            justifications_validators: BitList::with_capacity(
+                (num_validators * MAX_HISTORICAL_BLOCK_HASHES) as usize,
+            )
+            .expect("Failed to initialize an empty BitList"),
         }
     }
 
-    fn get_justifications_roots_index(&self, root: &B256) -> Option<usize> {
-        self.justifications_roots.iter().position(|r| r == root)
-    }
+    /// Returns a map of `root -> justifications` constructed from the
+    /// flattened data in the state.
+    pub fn get_justifications(&self) -> anyhow::Result<HashMap<B256, BitList<U4096>>> {
+        let mut justifications_map = HashMap::<B256, BitList<U4096>>::new();
```

Comment:
```suggestion
        let mut justifications_map = HashMap::new();
```

We don't need explicit type here.